src/spawn.rs

```diff
@@ -281,16 +277,16 @@ impl PosixSpawnFileActions {
     /// Add a [dup2](https://pubs.opengroup.org/onlinepubs/9699919799/functions/dup2.html) action. See
     /// [posix_spawn_file_actions_adddup2](https://pubs.opengroup.org/onlinepubs/9699919799/functions/posix_spawn_file_actions_adddup2.html).
     #[doc(alias("posix_spawn_file_actions_adddup2"))]
-    pub fn add_dup2<Fd1: AsFd, Fd2: AsFd>(
+    pub fn add_dup2<Fd1: AsRawFd, Fd2: AsRawFd>(
```

Comment:
Taking a look at the types that implement `AsRawFd`, they are mostly owned file descriptors, e.g., `File`, passing them by value would have them closed within these `add_xxx()` functions, and we don't have an impl like:


```rs
impl<T: AsRawFd> AsRawFd for &T
```


what about:

```rs
add_xxx<Fd: AsRawFd>(_: &AsRawFd)
```

Though one drawback with this interface is that when passing `RawFd`s, we still need to add a layer of reference to it:

```rs
let stdin_raw_fd: RawFd = 0;

add_xxx(&stdin_raw_fd);
```

thoughts on this?

Reply:
What about removing the generic type and just using `RawFd`?